### PR TITLE
[7zip] Add complete public header files

### LINKED ISSUE #27553: 7zip Please include 7z.h in the distribution for C developers
**Is your feature request related to a problem? Please describe.**
The main C header file `7z.h` for 7-Zip is missing which makes it impossible to use the vcpkg package on C projects.

This header file includes the primary functions used to interact with the library in C, such as:

- SzArEx_Init
- SzArEx_Free
- SzArEx_Open
- SzArEx_Extract

**Proposed solution**
Please include `include\7zip\C\7z.h` in the distribution 😄 

**Describe alternatives you've considered**
I don't really have a workaround for this sadly.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #27553 

Note: Add complete public header files

Usage test succeeded

### CODE REVIEW

**Review by dg0yt — COMMENTED** [review]
Did you consider using  `file(GLOB ...)`?

**Comment by Osyotr** [comment]
Most of these headers are internal and symbols inside them are not exported from shared library.

**Comment by FrankXie05** [comment]
@dg0yt @dg0yt `file(GLOB)` method is better, as [Osyotr](https://github.com/Osyotr) said, I'm looking for if these files should all be installed. 🧐

**Comment by Osyotr** [comment]
For my WIP cross-platform 7zip library wrapper it was enough to add just these files:
```
CPP/Common/MyCom.h
CPP/Common/MyGuidDef.h
```
Not sure about files in `C/` folder.

**Comment by FrankXie05** [comment]
@Osyotr Maybe I should just go and ask upstream about this.

**Comment by fgimian** [comment]
I do wonder if this requires more examination than I originally anticipated.  I think that perhaps the C code is entirely separate to the C++ code in this library and as far as I can see, the C code only provides LZMA decompression while the C++ code supports all archive formats.

I may be incorrect about this but if I'm understanding correctly, it may actually be worth splitting this package into two variants or perhaps just providing the C++ header files.

I've honestly struggled to really understand the way this is structured; compared to both libarchive and unrar, the 7-Zip SDK is significantly more complicated to consume.

Huge thanks for all your help!
Fotis

File: ports/7zip/portfile.cmake [hunk 1]
```diff
@@ -28,3 +28,5 @@ file(
     DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
     RENAME copyright
 )
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```
Comment (MonicaLiu0311) [inline ports/7zip/portfile.cmake]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")

```

Please leave a blank line at the end of the file.

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thanks!

**Pull request merged**